packages/openapi/src/decorators.ts

```diff
@@ -225,64 +241,100 @@ function omitUndefined<T extends Record<string, unknown>>(data: T): T {
   return Object.fromEntries(Object.entries(data).filter(([k, v]) => v !== undefined)) as any;
 }
 
-function validateIsUri(context: DecoratorContext, url: string, propertyName: string) {
-  try {
-    new URL(url);
-    return true;
-  } catch {
+/** Get TagsMetadata set with `@tagMetadata` decorator */
+const [getTagsMetadata, setTagsMetadata] = unsafe_useStateMap<
+  Type,
+  { [name: string]: TagMetadata }
+>(OpenAPIKeys.tagsMetadata);
+
+/**
+ * Decorator to add metadata to a tag associated with a namespace.
+ * @param context - The decorator context.
+ * @param entity - The namespace entity to associate the tag with.
+ * @param name - The name of the tag.
+ * @param tagMetadata - Optional metadata for the tag.
+ */
+export const tagMetadataDecorator: TagMetadataDecorator = (
+  context: DecoratorContext,
+  entity: Namespace,
+  name: string,
+  tagMetadata?: TypeSpecValue,
+) => {
+  // Check if the namespace is a service namespace
+  if (!entity.decorators.some((decorator) => decorator.decorator === $service)) {
     reportDiagnostic(context.program, {
-      code: "not-url",
+      code: "no-service-found",
+      format: {
+        namespace: entity.name,
+      },
       target: context.getArgumentTarget(0)!,
-      format: { property: propertyName, value: url },
     });
-    return false;
   }
-}
 
-function validateAdditionalInfoModel(context: DecoratorContext, typespecType: TypeSpecValue) {
-  const propertyModel = context.program.resolveTypeReference(
-    "TypeSpec.OpenAPI.AdditionalInfo",
-  )[0]! as Model;
+  // Retrieve existing tags metadata or initialize an empty object
+  const tags = getTagsMetadata(context.program, entity) ?? {};
 
-  if (typeof typespecType === "object" && propertyModel) {
-    const diagnostics = checkNoAdditionalProperties(
-      typespecType,
+  // Check for duplicate tag names
+  if (tags[name]) {
+    reportDiagnostic(context.program, {
+      code: "duplicate-tag",
+      format: { tagName: name },
+      target: context.getArgumentTarget(0)!,
+    });
+    return;
+  }
+
+  // Initialize metadata with the tag name
+  let metadata: any = { name };
```

Comment:
```suggestion
  let metadata: TagMetadata = { name };
```
and it also seems like you are not supposed to set name here